src/reflect/scala/reflect/internal/tpe/TypeMaps.scala

```diff
@@ -664,20 +664,31 @@ private[internal] trait TypeMaps {
   }
 
   /** A base class to compute all substitutions */
-  abstract class SubstMap[T](from: List[Symbol], to: List[T]) extends TypeMap {
-    // OPT this check was 2-3% of some profiles, demoted to -Xdev
-    if (isDeveloper) assert(sameLength(from, to), "Unsound substitution from "+ from +" to "+ to)
+  abstract class AbstractSubstMap[T >: Null] extends TypeMap {
+    protected def from: List[Symbol] = Nil
+    protected def to: List[T] = Nil
 
     private[this] var fromHasTermSymbol = false
     private[this] var fromMin = Int.MaxValue
     private[this] var fromMax = Int.MinValue
     private[this] var fromSize = 0
-    from.foreach {
-      sym =>
+
+    protected def reload(): Unit = {
```

Comment:
_Note_: The `SubstMap` class, previously, had upon class initialisation the `-Xdev` check, as well as the declaration and initialisation of the summary `var`. Class initialisation is fine as long as the values it depends upon are immutable. However, since we are introducing subclasses with mutable fields, we need a procedure that can be called by subclasses when those fields are modified.